### DISCUSSION TITLE
SFML on Termux

### DISCUSSION THREAD

**Discussion by yoshy3904** [termux/termux-app#2891] [Q&A]
I have compiled a sfml application with the sfml package in the x11-repo. It should open a 200x200 px window displaying a green circle.
```
#include <SFML/Graphics.hpp>

int main()
{
    sf::RenderWindow window(sf::VideoMode(200, 200), "SFML works!");
    sf::CircleShape shape(100.f);
    shape.setFillColor(sf::Color::Green);
    while (window.isOpen())
    {
        sf::Event event;
        while (window.pollEvent(event))
        {
            if (event.type == sf::Event::Closed)
                window.close();
        }
        window.clear();
        window.draw(shape);
        window.display();
    }
    return 0;
}
```
I have problems running the application. I set up an VNC-Server, set DISPLAY=:0, but when trying to run the application ./sfml-app I get the error:
**Error of failed request: BadMatch(invalid parameter attributes)
Major opcode of failed request: 1 (X_CreateWindow)
Serial number of failed request: 14
Current serial number in output stream: 16**

**Comment by sylirre** [comment]
Try setting up a basic [graphical environment](https://wiki.termux.com/wiki/Graphical_Environment) using `fluxbox` window manager. Some programs fail to run when raw X server is used.

**Reply by yoshy3904** [reply]
Changing the window manager did not help. I used termux-x11 which worked. Talked about this more in another [issue](https://github.com/termux/termux-packages/issues/10865)
